src/handlers/updateUserProfile.ts

```diff
@@ -0,0 +1,33 @@
+import Joi from '@hapi/joi'
+import * as express from 'express-async-router'
+
+import * as db from '../gateways/storage'
+import { ValidatedRequest } from '../middlewares/validator'
+
+export const schema = {
+  body: Joi.object<db.UserReadModel>({
+    discordId: Joi.string().required(),
+    discordDiscriminator: Joi.number().required(),
+    discordName: Joi.string().required(),
+    discordAvatar: Joi.string().required(),
+    permissions: Joi.number().required(),
+    preferedClan: Joi.string()
+      .valid('', 'Crab', 'Crane', 'Dragon', 'Lion', 'Phoenix', 'Scorpion', 'Unicorn')
+      .optional(),
+    jigokuName: Joi.string().optional(),
+  }),
+}
+
+export async function handler(req: ValidatedRequest<typeof schema>, res: express.Response) {
+  if (!req.params.id) {
+    res.status(400).send()
+    return
+  }
+  if (req.user?.d_id !== req.params.id) {
+    res.status(403).send()
+    return
+  }
+  const user = db.updateUser(req.body)
```

Comment:
After the changes to store the clan using references to the `clans` table, having clan as an empty string will not be valid. To handle that I think we have two main options:

- Accepts empty string as a request parameter, then transform `clan: ""` into `clan: null` in the handler before sending it to `db.updateUser`
- Do not accept empty string as a request parameter, accepting the clans + `null` (proper `null`, not `"null"`), then just pass ahead whatever we receive.

Considering we would use the clan ids for storage, we should probably receive `preferredClanId` as a number or `null` and then move forward from there.